### PR TITLE
Doc/CLN: pass pre-commit

### PR DESCRIPTION
**Pull request by luke396**
xref #3603

### CODE REVIEW

File: scripts/generate_mkdocs.py [hunk 1]
```diff
@@ -3,15 +3,11 @@
...
 import subprocess
-import sys
 from pathlib import Path
 from typing import NamedTuple
 
 import yaml
 
-if sys.version_info < (3, 9):
-    raise RuntimeError("You need at least python 3.9 to run this script")
-
```
Comment (luke396) [inline scripts/generate_mkdocs.py:11]: Especially here, it seems that they should not be automatically removed

Comment (charliermarsh) [reply scripts/generate_mkdocs.py:11]: Ah, I think this is "correct", but can we add `requires-python = ">=3.9"` to the `pyproject.toml` file in this directory? Right now it's defaulting to Python 3.10.

Comment (charliermarsh) [reply scripts/generate_mkdocs.py:11]: Gonna add and merge.

**Comment by onerandomusername** [comment]
How did this manage to make it to the default branch anyhow?

**Pull request merged**